src/packages/@ncigdc/components/Modals/ControlledAccess/ControlledAccessModal.js

```diff
@@ -1,134 +1,111 @@
+        selectedStudies,
+        setSelectedStudies,
+        studiesList,
+        user,
+        userAccessList,
+      })}
+      headings={getHeadings({
+        user,
+      })}
+      tableContainerStyle={{ maxHeight: 350 }}
+      tableId="controlled-access-table"
+      />
 
-  return (
-    <BaseModal
-      closeText="Cancel"
-      extraButtons={isAuth
-        ? (
-          <Button
-            disabled={exploreButtonDisabled}
-            onClick={handleModalSubmit}
-            >
-            Explore
-          </Button>
-        )
-        : (
-          <LoginButton keepModalOpen>
-            <Button>Login</Button>
-          </LoginButton>
-      )}
-      title="Explore Controlled & Open Data"
-      >
-      <CADevSettings
-        isFakeLoggedIn={isFakeLoggedIn}
-        setIsFakeLoggedIn={setIsFakeLoggedIn}
-        setUserAccessList={setUserAccessList}
-        userAccessList={userAccessList}
-        />
-      <CAMessage isAuth={isAuth} userAccessList={userAccessList} />
-      <EntityPageHorizontalTable
-        data={dataFormatted}
-        headings={getHeadings({
-          isAuth,
-          userAccessList,
-        })}
-        tableContainerStyle={{ maxHeight: 350 }}
-        tableId="controlled-access-table"
-        />
-      {isAuth || (
-        <RestrictionMessage
-          compact
-          faClass="fa-lock"
-          faStyle={{ color: '#773388' }}
-          fullWidth
-          leftAlign
-          title="The controlled data require dbGaP access"
-          >
-          <span>
-            If you don't have access, follow the instructions for
-            {' '}
-            <a href="https://gdc.cancer.gov/access-data/obtaining-access-controlled-data" rel="noopener noreferrer" target="_blank">obtaining access to controlled data</a>
-            .
-          </span>
-        </RestrictionMessage>
-      )}
-    </BaseModal>
-  );
-};
+    {!!user || (
+      <RestrictionMessage
+        compact
+        faClass="fa-lock"
+        faStyle={{ color: '#773388' }}
+        fullWidth
+        leftAlign
+        title="The controlled data require dbGaP access"
+        >
+        <span>
+          {'If you don\'t have access, follow the instructions for '}
+          <a href="https://gdc.cancer.gov/access-data/obtaining-access-controlled-data" rel="noopener noreferrer" target="_blank">obtaining access to controlled data</a>
+          .
+        </span>
+      </RestrictionMessage>
+    )}
+  </BaseModal>
+);
 
-export default enhance(ControlledAccessModal);
+export default compose(
+  setDisplayName('EnhancedControlledAccessModal'),
+  connect(state => ({
+    user: state.auth.user,
+  })),
+  withState(
+    'selectedStudies',
+    'setSelectedStudies',
+    ({
+      activeControlledPrograms,
+      userAccessList,
+    }) => (userAccessList.length === 1 ? userAccessList : activeControlledPrograms),
+    // TODO: this ^^^ is a placeholder for the "controlled" array of the CA response.
```

Comment:
@samrichca Having been sure not to remove logic that was there, I've gone looking deeper, and remembered I'd already added this placeholder foreseeing the stuff to be handled later.